demo

```diff
@@ -182,214 +202,254 @@ exitfunc() {
         retval=$?
-        if [[ $retval -ne 0 ]]; then
-            echo "Failed to upload the Artifact to the demo server. curl error code: " $retval
-            echo "Sleeping for 5 seconds before making another attempt..."
+        EXISTS_ERROR=1
+        if [[ $retval -eq 0 ]]; then
+            TENANT_ID=$(echo "$TENANT_ID" | tr -d '\r')
         else
-            break
+            TENANT_ID=
         fi
-        if [[ $RETRIES -ge $RETRY_LIMIT ]]; then
-            echo "Retried $RETRIES times without success. Giving up."
-            exit 1
+    else
+        docker exec \
+               ${DOCKER_COMPOSE_PROJECT_NAME}_mender-useradm_1 \
+               /usr/bin/useradm create-user \
+               --username=${USER} \
+               --password=${PASSWORD} \
+               > /dev/null
+        retval=$?
+        EXISTS_ERROR=5
+    fi
+    if [[ $retval -eq 0 ]]; then
+        PRINT_LOGIN_INFO=1
+    elif [[ $retval -eq $EXISTS_ERROR ]]; then
+        # If the user exists, skip uploading the Artifact
+        UPLOAD_ARTIFACT=0
+        PRINT_USER_EXISTS=1
+    else
+        echo "docker exec error: " $retval
+        exit $retval
+    fi
+}
+
+maybe_launch_enterprise_client() {
+    if [[ $ENTERPRISE -eq 1 ]] && [[ $CLIENT -eq 1 ]]; then
+        if [[ $(docker ps -q -f name=${DOCKER_COMPOSE_PROJECT_NAME}_mender-client_1 | wc -l) -gt 0 ]]; then
+            # If already launched, we don't need to do anything.
+            :
+        elif [[ -n "$TENANT_ID" ]]; then
+            TENANT_TOKEN=$(docker exec \
+                                  ${DOCKER_COMPOSE_PROJECT_NAME}_mender-tenantadm_1 \
+                                  /usr/bin/tenantadm get-tenant \
+                                  --id $TENANT_ID \
+                               | jq -r .tenant_token)
+            # Now that we have the tenant token we can enable the client.
+            EXTRA_FILES="$EXTRA_FILES $CLIENT_FILES"
+            TENANT_TOKEN=$TENANT_TOKEN docker-compose \
+                        -f docker-compose.yml \
+                        -f docker-compose.storage.minio.yml \
+                        -f docker-compose.demo.yml \
+                        -p ${DOCKER_COMPOSE_PROJECT_NAME} \
+                        $EXTRA_FILES \
+                        "${ARGS[@]}" -d mender-client
+        else
+            echo "WARNING: Ignoring request to launch the Mender client."
+            echo "In Enterprise mode, the client can only be launched the first time the server"
+            echo "is started, when the first user is created. If you wish to start from scratch,"
+            echo "replace \`up\` with \`down -v\` first to reset, then rerun."
         fi
-        RETRIES=$((RETRIES+1))
-        sleep 5
-    done
+    fi
+}
 
-    errout=$(jq '.error' <<< $cout)
+maybe_upload_artifact() {
+    if [[ $UPLOAD_ARTIFACT -eq 1 ]]; then
+
+        local RETRIES=0
+        local retval=0
+        until [[ "$JWT" != "" ]]; do
```

Comment:
I know this is not you (In fact I think it is me), but maybe change to `-n` for consistency?